In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
data=pd.read_csv('Sales_data_zone_wise.csv')

In [4]:
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


In [15]:
#To check the null values
data.isna().sum()

Month       0
Zone - A    0
Zone - B    0
Zone - C    0
Zone - D    0
dtype: int64

1.The mean sale generated by each zone

In [7]:
data.describe()

Zone - A      Zone - B      Zone - C      Zone - D
count  2.900000e+01  2.900000e+01  2.900000e+01  2.900000e+01
mean   1.540493e+06  1.755560e+06  1.772871e+06  1.842927e+06
std    2.619401e+05  1.683899e+05  3.331937e+05  3.750165e+05
min    1.128185e+06  1.527574e+06  1.237722e+06  1.234311e+06
25%    1.305972e+06  1.606010e+06  1.523308e+06  1.520406e+06
50%    1.534390e+06  1.740365e+06  1.767047e+06  1.854412e+06
75%    1.820196e+06  1.875658e+06  2.098463e+06  2.180416e+06
max    2.004480e+06  2.091194e+06  2.290580e+06  2.364132e+06

In [13]:
x=data.mean()
x

Zone - A    1.540493e+06
Zone - B    1.755560e+06
Zone - C    1.772871e+06
Zone - D    1.842927e+06
dtype: float64

*The mean sales generated by Zone A,Zone B,Zone C and Zone D are 1540493,1755560,1772871 and 1842927 respectively.

2).Total sales generated by all the zones for each month.

In [8]:
df=pd.DataFrame({'X':data['Zone - A'],'Y':data['Zone - B'],'Z':data['Zone - C'],'A':data['Zone - D']})
df1=df.sum(axis=1)
df2=pd.DataFrame({'Month':data['Month'],'Sum':df1})
df2

Month      Sum
0    Month - 1  7022544
1    Month - 2  7152303
2    Month - 3  6475939
3    Month - 4  8174449
4    Month - 5  5995328
5    Month - 6  7151387
6    Month - 7  7287108
7    Month - 8  7816299
8    Month - 9  6703395
9   Month - 10  7128210
10  Month - 11  7032783
11  Month - 12  6111084
12  Month - 13  5925424
13  Month - 14  7155515
14  Month - 15  5934156
15  Month - 16  6506659
16  Month - 17  7149383
17  Month - 18  7083490
18  Month - 19  6971953
19  Month - 20  7124599
20  Month - 21  7389597
21  Month - 22  7560001
22  Month - 23  6687919
23  Month - 24  7784747
24  Month - 25  6095918
25  Month - 26  6512360
26  Month - 27  6267918
27  Month - 28  7470920
28  Month - 29  6772277

*The total sale obtained in month-1 is 7022544 and for month-2 is 7152303 etc.The above table shows monthwise sales 
 irrespective of the Zones.

3).To check whether all the zones generate the same amount of sales.

H0:All the zones generate the same amount of sales
H1:All the zones generate different amount of sales

In [9]:
import scipy.stats as stats
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

In [16]:
fvalue,pvalue=stats.f_oneway(data['Zone - A'],data['Zone - B'],data['Zone - C'],data['Zone - D'])
print(fvalue,pvalue)

5.672056106843581 0.0011827601694503335


In [ ]:
*Here the P-value is 0.00118 which is lower than the significance level 0.05 so we reject the null hypothesis and 
 we get that all the zones generate different amount of sales. 